lib/api/request_adapter.rb

```diff
@@ -55,10 +61,13 @@ def hide?(thing)
     end
 
     def json_body
-      @json_body ||= begin
-                       body = @request.body.read if @request.body
-                       body.blank? || body == "null" ? {} : JSON.parse(body)
-                     end
+      @json_body ||=
+        begin
+          body = @request.body.read if @request.body
+          body.blank? || body == "null" ? {} : JSON.parse(body)
+        rescue JSON::ParserError
+          @request.params
```

Comment:
@agrare you ok with this change?

It is strange that http `OPTIONS` does not have a JSON `body` while `GET` and `POST` seem to handle the parameters just fine.

Reply:
I was hoping we'd be able to find out why it is different since it seems strange, e.g. is it a default Content-Type thing?

Reply:
ok. not sure what was up with rack, but it is strange that they are reporting that there is a body. A body is not allowed from the spec. It could be in rack or a bug in our http request code